client/packages/purchase_orders/src/api/hooks/usePurchaseOrder.ts

```diff
@@ -140,3 +146,59 @@ const useFilteredAndSortedLines = (
 
   return { sortedAndFilteredLines, itemFilter, setItemFilter };
 };
+
+const useAddFromMasterList = () => {
+  const { purchaseOrderApi, storeId, queryClient } = usePurchaseOrderGraphQL();
+  const t = useTranslation();
+  const { error } = useNotification();
+  
+
+  const getConfirmation = useConfirmationModal({
+    title: t('heading.are-you-sure'),
+    message: t('messages.confirm-add-from-master-list'),
+  });
+
+  const mutationState =  useMutation(purchaseOrderApi.addToPurchaseOrderFromMasterList, {
+    onSettled: () =>
+      queryClient.invalidateQueries([LIST, PURCHASE_ORDER, storeId]),
+  });
+
+  const addFromMasterList = async (masterListId: string, purchaseOrderId: string) => {
+    getConfirmation({
+      onConfirm: async () => {
```

Comment:
error handling and adding useConfirm within the hook, but I was wondering if this should be in the button component?